client.go

```diff
@@ -18,10 +20,22 @@ type ldapClient interface {
 	Execute(conf *client.Config, entries []*ldif.Entry, continueOnError bool) (err error)
 }
 
-func NewClient() *Client {
-	return &Client{
+func NewClient(conf *logical.BackendConfig) *Client {
```

Comment:
We probably don't need the whole config and could just accept an `hclog.Logger`.